File: src/main/java/com/tms/csp/transforms/RmpsToOr.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;

/**
 * Replaces each imp(a b) with an or(!a b)
 */
public class RmpsToOr extends BaseTransformer {

    @Override
    protected Exp local(Exp in) {
        return in.asRmp().toOr();
    }

    @Override
    protected boolean executeLocal(Exp in) {
        return in.isRmp();
    }


}

File: src/main/java/com/tms/csp/transforms/RmpsToOr.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

/**
 * Replaces each imp(a b) with an or(!a b)
 */
class RmpsToOr : BaseTransformer() {

    override fun local(inExp: Exp): Exp? {
        return inExp.asRmp().toOr
    }

    override fun executeLocal(expIn: Exp): Boolean {
        return expIn.isRmp
    }


}
